app/services/articles/feeds/large_forem_experimental.rb

```diff
@@ -126,26 +126,74 @@ def score_comments(article)
-          .order(hotness_score: :desc)
-        featured_story = hot_stories.where.not(main_image: nil).first
         if user_signed_in
-          hot_story_count = hot_stories.count
-          offset = RANDOM_OFFSET_VALUES.select do |i|
-            i < hot_story_count
-          end.sample # random offset, weighted more towards zero
-          hot_stories = hot_stories.offset(offset)
+          hot_stories = experimentalal_hot_story_grab
+          featured_story = hot_stories.where.not(main_image: nil).first
           new_stories = Article.published
             .where("score > ?", -15)
             .limited_column_select.includes(top_comments: :user).order(published_at: :desc).limit(rand(15..80))
           hot_stories = hot_stories.to_a + new_stories.to_a
+        else
+          hot_stories = Article.published.limited_column_select
+            .page(@page).per(@number_of_articles)
+            .where("score >= ? OR featured = ?", SiteConfig.home_feed_minimum_score, true)
+            .order(hotness_score: :desc)
+          featured_story = hot_stories.where.not(main_image: nil).first
         end
         [featured_story, hot_stories.to_a]
       end
 
       private
 
+      # Disable complexity cop to allow for variant-driven method
+      # rubocop:disable Metrics/CyclomaticComplexity
+      def experimentalal_hot_story_grab
+        test_variant = @user ? field_test(:feed_top_articles_query, participant: @user) : "base"
+        case test_variant
+        when "base_with_more_articles" # equivalent to current base but with higher "number of articles"
+          articles = Article.published.limited_column_select.includes(top_comments: :user)
+            .page(@page).per(75)
+            .where("score >= ? OR featured = ?", SiteConfig.home_feed_minimum_score, true)
+            .order(hotness_score: :desc)
+        when "only_followed_tags" # equivalent to base but only on tags user follows (if user follows enough)
+          followed_tags = @user.cached_followed_tag_names
+          articles = Article.published.limited_column_select.includes(top_comments: :user)
+            .page(@page).per(@number_of_articles)
+            .where("score >= ? OR featured = ?", SiteConfig.home_feed_minimum_score, true)
+            .order(hotness_score: :desc)
+          # We only want to limit the posts to tagged_with if the participant follows enough tags.
+          articles = articles.tagged_with(followed_tags, any: true) if followed_tags.size > 4
+        when "top_articles_since_last_pageview_3_days_max" # Top articles since last page view (max 3 days)
+          start_time = [(@user.page_views.last&.created_at || Time.current) - 12.hours, 3.days.ago].max
+          articles = Article.published.limited_column_select.includes(top_comments: :user)
+            .where("published_at > ?", start_time)
+            .page(@page).per(@number_of_articles)
+            .order(score: :desc)
```

Comment:
Why are some of these ordered by `hotness_score` and others by `score`?

Reply:
"score" is the overall ranking of a post, regardless of when it was created. "hotness" takes into account the age of the post. So if we are not selecting for a time period, "hotness score" works because it has a bias towards recency.

This PR proposes some alternatives to using hotness score which selects _just_ for score, but cuts off how recent based on when the person was most recently active (based on their last page view)... So if you haven't been to the site in a few days, you'll get top posts instead of just current "hot" ones. But if you have been checking back more often you get the more recent top posts. This caters the feed a little more to the browsing habits of the user...

I have no idea if this is going to be more relevant and engaging, but we'll have some results in a few weeks to see. I think new complexity is isolated and will be pulled at the end of the trial.